e3nn/tensor/cartesian_tensor.py

```diff
@@ -0,0 +1,28 @@
+import torch
+from e3nn.tensor.irrep_tensor import IrrepTensor
+from e3nn import rs, o3
+
+
+class CartesianTensor():
+    def __init__(self, tensor, formula=None):
+        self.N = len(tensor.shape)
+        if self.N > 13:
+            raise ValueError("CartesianTensor only supports up to 13 indices.")
```

Comment:
I would remove this but check below that len(formula) is equal to tensor.dim()

Reply:
ah indeed! of course. thanks!